main.go

```diff
@@ -61,26 +62,46 @@ func main() {
 	flag.StringVar(&leaderElectionResourceLock, "leader-election-resource", "configmaps", "determines which resource lock to use for leader election. option:[configmapsleases|endpointsleases|configmaps]")
 
 	// Custom flags
-	var printVersion, pprofActive bool
+	var printVersion, pprofActive, ddProfilingEnabled bool
 	var logEncoder string
 	flag.StringVar(&logEncoder, "logEncoder", "json", "log encoding ('json' or 'console')")
 	logLevel := zap.LevelFlag("loglevel", zapcore.InfoLevel, "Set log level")
 	flag.BoolVar(&printVersion, "version", false, "Print version and exit")
 	flag.BoolVar(&pprofActive, "pprof", false, "Enable pprof endpoint")
+	flag.BoolVar(&ddProfilingEnabled, "ddProfilingEnabled", false, "Enable the datadog profiler")
 
 	// Parsing flags
 	flag.Parse()
 
+	exitCode := 0
+	defer func() { os.Exit(exitCode) }()
+
+	if ddProfilingEnabled {
+		setupLog.Info("Starting datadog profiler")
+		if err := profiler.Start(
```

Comment:
Would it be useful to add `profiler.WithService("eds")` or similar to be able to filter later?

Reply:
This needs to stay configurable depending on how users want to name the service, for internal usage I'm planning to define `DD_SERVICE` in the eds pod, it will be read and applied by the profiler library.